stdlib/format.mli

```diff
@@ -1405,3 +1407,141 @@ val kasprintf : (string -> 'a) -> ('b, formatter, unit, 'a) format4 -> 'b
+  val pp_pair : Format.formatter -> int * bool -> unit = <fun>
+  # Format.printf "l: [@[<hov>%a@]]@."
+    Format.(pp_print_list ~pp_sep:(fun out () -> fprintf out ";@ ") pp_pair) l
+    l: [(0, true); (1, false); (2, true); (3, false); (4, true); (5, false);
+       (6, true); (7, false); (8, true); (9, false); (10, true); (11, false);
+       (12, true); (13, false); (14, true); (15, false); (16, true);
+       (17, false); (18, true); (19, false)]
+
+  ]}
+
+
+  What this does, briefly, is:
+
+    - [pp_pair] prints a pair [bool*int] surrounded in "(" ")". It takes
+      a formatter (into which formatting happens), and the pair itself.
+      When printing is done it returns [()].
+
+    - [Format.printf "l = [@[<hov>%a@]]@." ... l] is like [printf], but
+      with additional formatting instructions (denoted with "@"). The pair
+      "@[<hov>" and "@]" is an "horizontal-or-vertical box".
+
+    - "@." ends formatting with a newline. It is similar to "\n" but is also
+      aware of the [Format.formatter]'s state. Do not use "\n" with [Format].
+
+    - "%a" is a formatting instruction, like "%d" or "%s" for [printf].
+      However, where "%d" prints an integer and "%s" prints a string,
+      "%a" takes a printer (of type [Format.formatter -> 'a -> unit])
+      and a value (of type ['a]) and applies the printer to the value.
+      This is key to compositionality of printers.
+
+    - We build a list printer using
+      [Format.pp_print_list ~pp_sep:(...) pp_pair].
+      [pp_print_list] takes an element printer and returns a list printer.
+      The [?pp_sep] optional argument, if provided, is called in between
+      each element to print a separator.
+
+    - Here, for a separator, we use [(fun out () -> Format.fprintf out ";@ ")].
+      It prints ";", and then "@ " which is a breaking space
+      (either it prints " ", or it prints a newline if the box is about to
+      overflow).
+      This "@ " is responsible for the list printing splitting into several
+      lines.
+
+  If we omit "@ ", we get an ugly single-line print:
+
+  {[# Format.printf "l: [@[<hov>%a@]]@."
+      Format.(pp_print_list ~pp_sep:(fun out () -> fprintf out "; ") pp_pair) l
+  l: [(0, true); (1, false); (2, true); (* ... *); (18, true); (19, false)]
+- : unit = ()
+    ]}
+
+  Generally, it is good practice to define custom printers for important types
+  in your program. If, for example, you were to define basic geometry
+  types like so:
+
+  {[
+  type point = {
+    x: float;
+    y: float;
+  }
+
+  type rectangle = {
+    ll: point; (* lower left *)
+    ur: point; (* upper right *)
+  }
+  ]}
+
+  It would be convenient, for debugging purpose, or to display information
+  in logs, or on the console, to define printers. Here is an example of to do
+  it. Note that "%.3f" is a [float] printer up to 3 digits of precision
+  after the dot; "%f" would print as many digits as required, which is
+  somewhat verbose; "%h" is an hexadecimal float printer.
```

Comment:
```suggestion
  For debugging purposes, or to display information
  in logs or the console, it would be convenient to define printers for these types. Here is an example of how to do
  it. Note that "%.3f" is a [float] printer with 3 digits of precision
  after the dot; "%f" would print as many digits as required, which is
  somewhat verbose; "%h" is a hexadecimal float printer.
```